// @splitsoftware/splitio-react/lib/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
exports.__esModule = true;
exports.SplitContext = exports.useManager = exports.useTrack = exports.useTreatments = exports.useClient = exports.SplitFactory = exports.SplitClient = exports.SplitTreatments = exports.withSplitTreatments = exports.withSplitClient = exports.withSplitFactory = exports.SplitSdk = void 0;
// Split SDK factory (Renamed to avoid name conflict with SplitFactory component)
var splitio_1 = require("./splitio");
__createBinding(exports, splitio_1, "SplitFactory", "SplitSdk");
// HOC functions
var withSplitFactory_1 = require("./withSplitFactory");
__createBinding(exports, withSplitFactory_1, "default", "withSplitFactory");
var withSplitClient_1 = require("./withSplitClient");
__createBinding(exports, withSplitClient_1, "default", "withSplitClient");
var withSplitTreatments_1 = require("./withSplitTreatments");
__createBinding(exports, withSplitTreatments_1, "default", "withSplitTreatments");
// Render props components
var SplitTreatments_1 = require("./SplitTreatments");
__createBinding(exports, SplitTreatments_1, "default", "SplitTreatments");
var SplitClient_1 = require("./SplitClient");
__createBinding(exports, SplitClient_1, "default", "SplitClient");
var SplitFactory_1 = require("./SplitFactory");
__createBinding(exports, SplitFactory_1, "default", "SplitFactory");
// helper functions/hooks
var useClient_1 = require("./useClient");
__createBinding(exports, useClient_1, "default", "useClient");
var useTreatments_1 = require("./useTreatments");
__createBinding(exports, useTreatments_1, "default", "useTreatments");
var useTrack_1 = require("./useTrack");
__createBinding(exports, useTrack_1, "default", "useTrack");
var useManager_1 = require("./useManager");
__createBinding(exports, useManager_1, "default", "useManager");
// SplitContext
var SplitContext_1 = require("./SplitContext");
__createBinding(exports, SplitContext_1, "default", "SplitContext");
